### DISCUSSION TITLE
Image Transparency

### DISCUSSION THREAD

**Discussion by LCagmac** [DigiScore/neoscore#97] [Q&A]
Hi! I saw that there is information about using alpha transparency with brushes and pens in the documentation. I wanted to find out if this is possible with images ( this is for the neoscore competition score creation), and how I could go about doing this.

Thank you!

**Comment by ajyoon** [comment]
I think we don't support that right now, but it should be easy to add. I'll take a look this week.

Andrew Yoon | andrewyoon.art

-------- Original Message --------

**Reply by ajyoon** [reply]
Ok, it was an easy improvement so [I just pushed support for this](https://github.com/DigiScore/neoscore/commit/f3608524d65871ba1d8d6761e7c274b70b3f596f). It'll come out in 0.1.12, which I expect will be some time this week.

**Reply by LCagmac** [reply]
Thank you so much! I look forward to checking out this feature later this week

**Reply by ajyoon** [reply]
0.1.12 is out now! See `Image.opacity`

**Reply by LCagmac** [reply]
Great! Thank you!

Also (sorry for asking so many questions), is there a way to change a shape's color after it has been drawn? I see that I can change the background with my code over time, but I have just had some challenges with changing shape's colors.

**Reply by ajyoon** [reply]
No apologies needed, questions are great! You can change the colors of paths by changing the brush (for fill colors/patterns) and pen (for outline) at runtime (typically inside the main refresh function, assuming you're using one already). Something like..

```py
from neoscore.common import *

neoscore.setup()

shape = Path.rect(ORIGIN, None, Mm(50), Mm(30))

def refresh_func(time):
    # For demonstration purpose, change the brush's red channel with time.
    # In practice you can do anything here to make new brushes.
    shape.brush = Brush((int((time * 50) % 255), 0, 0))

neoscore.show(refresh_func)
```

**Reply by ajyoon** [reply]
if you're talking about changing the colors of _images_ at runtime, that is not currently supported well. There are some ways you could do it by managing the image in an image manipulation library like Pillow, then deleting the neoscore Image and replacing it with a new one derived from the edited image, but this is kind of hacky and easy to do wrong, so I recommend working with natively supported `Path`s when possible.

**Reply by ajyoon** [reply]
btw - paths support opacity already, since brushes and pens can take an alpha channel (RGBA instead of just RGB). see https://neoscore.org/core_concepts/brushes_and_pens.html#colors

**Reply by LCagmac** [reply]
Thank you!
